dbos/_context.py

```diff
@@ -374,6 +421,9 @@ def __enter__(self) -> DBOSContext:
             ctx = DBOSContext()
             _set_local_dbos_context(ctx)
         assert not ctx.is_within_workflow()
+        # The workflow timeout should not be set within the workflow
```

Comment:
What about child workflows or enqueued workflows? I think it'll be helpful if they can have a timeout as well.

Reply:
They can, and there are tests for that. But the parent's timeout isn't automatically propagated to its children.

Reply:
Now, maybe it should be, but we'll have to carefully think through the semantics of that.